src/components/ContentOpenWith/OpenWithButton.js

```diff
@@ -29,32 +29,53 @@ type Props = {
  * @private
  * @return {?(string | Element)} the tooltip message
  */
-export const getTooltip = (displayDescription: ?string, error: ?any, isLoading: boolean): ?(string | Element) => {
+export const getTooltip = (
+    isLoading: boolean,
+    error: ?Error,
+    isDisabled: boolean,
+    disabledReasons: Array<string> = [],
+    displayDescription: string,
+): ?(string | Element) => {
     if (isLoading) {
         return null;
     }
+
     if (error) {
-        return <FormattedMessage {...messages.errorOpenWithDescription} />;
+        return disabledReasons[0] || <FormattedMessage {...messages.errorOpenWithDescription} />;
+    }
+
+    if (isDisabled) {
+        return disabledReasons[0] || <FormattedMessage {...messages.emptyOpenWithDescription} />;
     }
+
     if (displayDescription) {
         return displayDescription;
     }
 
-    return <FormattedMessage {...messages.emptyOpenWithDescription} />;
+    return disabledReasons[0] || <FormattedMessage {...messages.emptyOpenWithDescription} />;
```

Comment:
I would reduce the number of returns here specially considering some are returning something similar. The lesser the number of exit points from your function, less break points you/someone will have to put to debug what the final value ends up being. You could just figure out what message to show and at the end 

if (error) message = messages.errorOpenWithDescription;
else if (isDisabled) message = messages.emptyOpenWithDescription;
else message = messages.emptyOpenWithDescription

`return disabledReasons[0] || <FormattedMessage {...message} />`